### PR TITLE
Update derived column migration to prevent conflicting column names and prevent duplicate column names

### PR DESCRIPTION
**Pull request by nllong**
#### Any background context you want to provide?
* See the original issue from @ebeers-png #3439 
* This update makes us that *maybe* forcing column names to be unique via a database constraint would be a good idea. Uniqueness constraint would be a combination of `org`, `table_name`, and `column_name`

#### What's this PR do?
* updates the migration to check if the creation of the column from the derived column will result in a name conflict
* adds in the `is_extra_data` to the column creations per the original issue's suggestion

#### How should this be manually tested?
Testing is a bit complicated since most databases have probably already run this migration. I've documented my testing in case you don't want to test this workflow.

* Checkout SEED to a version before the `0170` migration existed (I used sha `d43fdfc2e`)
* In a clean database since you went back to a pretty old commit, import some data
* Create a derived column with the same name as an existing column

image

* verify that you can view the two same-named columns in the property list
image

* check out this branch and run the migrations
* go to the column list profile and add in the renamed derived columns (has "derived 1" appended).
* Verify the column is correctly displayed in the inventory list

image

* Also verify that you can no longer create a derived column with an existing column name (e.g., lot_number). 

image

#### What are the relevant tickets?
#3439

### CODE REVIEW

**Comment by nllong** [comment]
@ebeers-png  -- do you want to check this PR out too?

File: seed/tests/test_derived_columns.py [hunk 1]
```diff
@@ -639,34 +639,49 @@ def test_derived_column_evaluation_with_derived_column_as_source_column(self):
         # -- Setup
         # expression which sums all the parameters
         expression = '$a + 2'
+        derived_column_name = 'dc1'
```
Comment (haneslinger) [inline seed/tests/test_derived_columns.py:642]: does this break if you try to get two derived columns?

**Review by haneslinger — APPROVED** [review]

**Comment by ebeers-png** [comment]
> @ebeers-png -- do you want to check this PR out too?

It looks great, thank you for all the thorough testing!

**Comment by nllong** [comment]
> > @ebeers-png -- do you want to check this PR out too?
> 
> It looks great, thank you for all the thorough testing!

Thanks for confirming @ebeers-png

**Pull request merged**